meta-iotconnect/recipes-apps/temp/files/0002-app-config.patch

```diff
@@ -0,0 +1,59 @@
+From 982095bca4f90688c3387f6aae602ccddc85b3e2 Mon Sep 17 00:00:00 2001
+From: "P.Young" <pyoung@witekio.com>
+Date: Tue, 18 Apr 2023 09:28:07 +0100
+Subject: [PATCH] changed preprocessors to allow for compiler argument
+ over-ride
+
+---
+ samples/basic-sample/config/app_config.h | 21 ++++++++++++++++++---
+ 1 file changed, 18 insertions(+), 3 deletions(-)
+
+diff --git a/samples/basic-sample/config/app_config.h b/samples/basic-sample/config/app_config.h
+index b70c45f..fc3bdd5 100644
+--- a/samples/basic-sample/config/app_config.h
++++ b/samples/basic-sample/config/app_config.h
+@@ -7,23 +7,38 @@
+ 
+ #include "iotconnect.h"
+ 
+-#define IOTCONNECT_CPID "your-cpid"
+-#define IOTCONNECT_ENV  "your-enviroment"
++#ifndef IOTCONNECT_CPID
```

Comment:
Why not just allow the user to define these values in the file in the way it is right now?
```
#ifndef COMPILER_FLAG
#define COMPILER_FLAG "your-whateverdefault"
#endif
```
And then at runtime you can detect with strcmp if COMPILER_FLAG define is set to "your-whateverdefault", and then error out there.

Then these changes can be to be applied to the generic sdk repo.

Reply:
The changes that I am proposing are for the main repo and they  seem to fit your principle of don't edit sources within yocto. Am I not understanding what you are saying or you are not understanding what I am saying.

What I am suggesting is:
```c
#ifndef IOTCONNECT_CPID 
#define IOTCONNECT_CPID "your-cpid"
#endif
```

And then you add -DIOTCONNECT_CPID=mycpid to your bitbake recipe's c flags.

What really should be happening here is that customer would have their own application and sample_app.c would not be even compiled in. They would not most likely have defines for CPID/DUID etc. but a config would be read from a file at runtime. So all this discussion is moot in my opinion.

This change is not crucial. I will approve the review if you don't implement this.

Reply:
The point that I was trying to make is that I have hoped that you would adapt and improve app_config.h in the original repo.

I am dropping this issue and acceptring your resotuion. Feel free to revert back to the original patch approach, if youi feel that it is better than adding the app_config.h to bbappend.